### PR TITLE
Lynx 12.13.0 plat 7136

### PR DESCRIPTION
**Pull request by shpiNi**

### CODE REVIEW

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php [hunk 1]
```diff
@@ -175,6 +202,30 @@ public function __construct($feedId, $feedProcessingKey = null, $ks = null)
 		KalturaLog::info("syndicationFeedRenderer- initialization done [".($microTimeEnd - $microTimeStart)."]");		
 	}
 	
+	private function extractStateParams($state)
+	{
+		$decodedState = base64_decode($state);
+		list($lastCreatedAt, $excludedEntryIdsStr) = explode(",", $decodedState);
```
Comment (rkreich) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: Check if string contains `","`

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php [hunk 2]
```diff
@@ -175,6 +202,30 @@ public function __construct($feedId, $feedProcessingKey = null, $ks = null)
 		KalturaLog::info("syndicationFeedRenderer- initialization done [".($microTimeEnd - $microTimeStart)."]");		
 	}
 	
+	private function extractStateParams($state)
+	{
+		$decodedState = base64_decode($state);
```
Comment (rkreich) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: Check decoded data `!== false`

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php [hunk 3]
```diff
@@ -189,6 +240,16 @@ public function addFlavorParamsAttachedFilter()
...
 	}
 	
+	public function addExternalAttachedFilter()
+	{
+		$entryFilter = $this->syndicationFeed->entryFilter;
+
+		$coreFilter = new entryFilter();
+		$entryFilter->toObject($coreFilter);
+		$entryFilter = $coreFilter;
+		$this->addFilter($entryFilter);
```
Comment (rkreich) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: Will `$this->baseCriteria` be the base criteria in this case as well?

Comment (shpiNi) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: yes, the baseentry filter will be attached to it.

File: api_v3/lib/syndication/KalturaFeedRenderer.php [hunk 4]
```diff
@@ -6,6 +6,8 @@ class KalturaFeedRenderer extends SyndicationFeedRenderer{
 	
 	protected $kalturaXslt = null;
 	protected $kalturaXsltItem = null;
+
+	private $state;
```
Comment (MosheMaorKaltura) [inline api_v3/lib/syndication/KalturaFeedRenderer.php]: State should be assign with default value.

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php [hunk 5]
```diff
@@ -142,8 +152,26 @@ public function __construct($feedId, $feedProcessingKey = null, $ks = null)
...
+				list($lastCreatedAt, $excludedEntryIds) = $this->extractStateParams($state);
+
+				$this->baseCriteria->add(entryPeer::CREATED_AT, $lastCreatedAt, Criteria::LESS_EQUAL);
+				$this->baseCriteria->addAnd(entryPeer::ID, $excludedEntryIds, Criteria::NOT_IN);
+			}
+			entryPeer::allowDeletedInCriteriaFilter();
+		}
+		else
+		{
+			$this->baseCriteria->addAnd(entryPeer::STATUS, entryStatus::READY);
```
Comment (MosheMaorKaltura) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: This line is not needed- remove.

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php [hunk 6]
```diff
@@ -499,11 +581,16 @@ public function execute($limit = 0)
...
 				}
-			}
-			
+			} 
 			echo $renderer->finalize($xml, $nextEntry !== false);
 		}
 		
+		if($this->addLinkForNextIteration)
+		{
+			$currState = $this->encodeStateParams($currentLastCreatedAtVal, $excludedEntryIds);
```
Comment (MosheMaorKaltura) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: $currentLastCreatedAtVal will not be known in this scope

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php [hunk 7]
```diff
@@ -499,11 +581,19 @@ public function execute($limit = 0)
...
+			} 
+
+			
 			
 			echo $renderer->finalize($xml, $nextEntry !== false);
 		}
 		
+		if($this->addLinkForNextIteration)
+		{
+			$currState = $this->encodeStateParams($currentLastCreatedAtVal, $excludedEntryIds);
```
Comment (shpiNi) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: done

File: api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php (hunk 6 above)
Comment (shpiNi) [inline api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php]: done, declared var outside of while loop

**Pull request merged**